lib/galaxy/tools/deps/conda_util.py

```diff
@@ -82,6 +82,39 @@ def __init__(self, conda_prefix=None, conda_exec=None,
             ensure_channels = None
         self.ensure_channels = ensure_channels
         self.ensured_channels = False
+        self._conda_version = None
+        self._miniconda_version = None
+
+    @property
+    def conda_version(self):
+        if self._conda_version is None:
+            self._guess_conda_version()
+        return self._conda_version
+
+    def _guess_conda_version(self):
+        conda_meta_path = self._conda_meta_path
+        # Perhaps we should call "conda info --json" and parse it but for now we are going
+        # to assume the default.
+        conda_version = LooseVersion(CONDA_VERSION)
+        miniconda_version = "3"
+
+        if os.path.exists(conda_meta_path):
+            for package in os.listdir(conda_meta_path):
+                package_parts = package.split("-", 2)
+                if len(package_parts) < 3:
+                    continue
+                package, version, build = package_parts
```

Comment:
Since the package name can contain dashes, this may result in parsing e.g. `conda-build-2.0.11-py27_0.json` as `package = 'conda', version = 'build', build = '2.0.11-py27_0.json'` .

I think a better solution would be:
```python
                package_parts = package.split("-")
                if len(package_parts) < 3:
                    continue
                package = '-'.join(package_parts[:-2])
                version = package_parts[-2]
                build = package_parts[-1]
```